testkit/src/main/scala/org/bitcoins/testkit/lnd/LndRpcTestUtil.scala

```diff
@@ -0,0 +1,344 @@
+                                        zmqPort = zmqPort)
+      case BitcoindVersion.V16 =>
+        BitcoindRpcTestUtil.v16Instance(port = port,
+                                        rpcPort = rpcPort,
+                                        zmqPort = zmqPort)
+      case BitcoindVersion.Experimental =>
+        BitcoindRpcTestUtil.vExperimentalInstance(port = port,
+                                                  rpcPort = rpcPort,
+                                                  zmqPort = zmqPort)
+      case BitcoindVersion.Unknown =>
+        sys.error(s"Cannot start lnd with an unknown instance of bitcoind")
+    }
+  }
+
+  def commonConfig(
+      bitcoindInstance: BitcoindInstance,
+      port: Int = RpcUtil.randomPort,
+      rpcPort: Int = RpcUtil.randomPort): String = {
+    val rawTx = bitcoindInstance.zmqConfig.rawTx.get
+    val rawBlock = bitcoindInstance.zmqConfig.rawBlock.get
+    s"""
+       |noseedbackup = true
+       |bitcoin.active = true
+       |bitcoin.regtest = true
+       |bitcoin.node = bitcoind
+       |norest=true
+       |debuglevel=critical
+       |listen=127.0.0.1:$port
+       |rpclisten=127.0.0.1:$rpcPort
+       |bitcoind.rpcuser = ${bitcoindInstance.authCredentials
+      .asInstanceOf[BitcoindAuthCredentials.PasswordBased]
+      .username}
+       |bitcoind.rpcpass = ${bitcoindInstance.authCredentials
+      .asInstanceOf[BitcoindAuthCredentials.PasswordBased]
+      .password}
+       |bitcoind.rpchost = 127.0.0.1:${bitcoindInstance.rpcUri.getPort}
+       |bitcoind.zmqpubrawtx = tcp://127.0.0.1:${rawTx.getPort}
+       |bitcoind.zmqpubrawblock = tcp://127.0.0.1:${rawBlock.getPort}
+       |""".stripMargin
+  }
+
+  def lndDataDir(
+      bitcoindRpcClient: BitcoindRpcClient,
+      isCannonical: Boolean): File = {
+    val bitcoindInstance = bitcoindRpcClient.instance
+    if (isCannonical) {
+      //assumes that the ${HOME}/.lnd/lnd.conf file is created AND a bitcoind instance is running
+      cannonicalDatadir
+    } else {
+      //creates a random lnd datadir, but still assumes that a bitcoind instance is running right now
+      val datadir = randomLndDatadir()
+      datadir.mkdirs()
+      logger.trace(s"Creating temp lnd dir ${datadir.getAbsolutePath}")
+
+      val config = commonConfig(bitcoindInstance)
+
+      new PrintWriter(new File(datadir, "lnd.conf")) {
+        write(config)
+        close()
+      }
+      datadir
+    }
+  }
+
+  def lndInstance(bitcoindRpc: BitcoindRpcClient): LndInstance = {
+    val datadir = lndDataDir(bitcoindRpc, isCannonical = false)
+    lndInstance(datadir)
+  }
+
+  def lndInstance(datadir: File): LndInstance = {
+    LndInstance.fromDataDir(datadir)
+  }
+
+  /** Returns a `Future` that is completed when both lnd and bitcoind have the same block height
+    * Fails the future if they are not synchronized within the given timeout.
+    */
+  def awaitLndInSync(lnd: LndRpcClient, bitcoind: BitcoindRpcClient)(implicit
```

Comment:
Not related to this PR, but I wanted to leave a note to re-visit this when we address #2844 

It would be very nice to have abstractions for this so we don't have to write new helper methods to do it

Reply:
Really the rest of the file